modules/core/shared/src/main/scala/eu/timepit/refined/types/net.scala

```diff
@@ -23,4 +25,61 @@ trait NetTypes {
 
   /** An `Int` in the range from 1024 to 65535 representing a port number. */
   type NonSystemPortNumber = Int Refined Interval.Closed[W.`1024`.T, W.`65535`.T]
+
+  import PrivateNetworks._
+
+  /** A `String` representing a valid IPv4 in the private network 10.0.0.0/8 (RFC1918) */
+  type Rfc1918ClassAPrivate = String Refined Rfc1918ClassAPrivateSpec
+
+  /** A `String` representing a valid IPv4 in the private network 172.15.0.0/12 (RFC1918) */
+  type Rfc1918ClassBPrivate = String Refined Rfc1918ClassBPrivateSpec
+
+  /** A `String` representing a valid IPv4 in the private network 192.168.0.0/16 (RFC1918) */
+  type Rfc1918ClassCPrivate = String Refined Rfc1918ClassCPrivateSpec
+
+  /** A `String` representing a valid IPv4 in a private network according to RFC1918 */
+  type Rfc1918Private =
+    String Refined Rfc1918ClassAPrivateSpec Or Rfc1918ClassBPrivateSpec Or Rfc1918ClassCPrivateSpec
```

Comment:
Would it make sense to add an alias `type Rfc1918PrivateSpec = Rfc1918ClassAPrivateSpec Or Rfc1918ClassBPrivateSpec Or Rfc1918ClassCPrivateSpec`? That could be used here and in the definition of `PrivateNetwork`. A `type Rfc5737TestnetSpec` could be also useful to avoid repetition.